src/adagrams.js

```diff
@@ -1,15 +1,146 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+class Adagrams {
+  static drawLetters = () => {
+    // Implement this method for wave 1
+    const letterPool = {
+      A: 9,
+      B: 2,
+      C: 2,
+      D: 4,
+      E: 12,
+      F: 2,
+      G: 3,
+      H: 2,
+      I: 9,
+      J: 1,
+      K: 1,
+      L: 4,
+      M: 2,
+      N: 6,
+      O: 8,
+      P: 2,
+      Q: 1,
+      R: 6,
+      S: 4,
+      T: 6,
+      U: 4,
+      V: 2,
+      W: 2,
+      X: 1,
+      Y: 2,
+      Z: 1,
+    };
 
-export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+    let letterList = [];
 
-export const scoreWord = (word) => {
-  // Implement this method for wave 3
-};
+    for (const [key, value] of Object.entries(letterPool)) {
+      for (let i = 0; i < value; i++) {
+        letterList.push(key);
+      }
+    }
 
-export const highestScoreFrom = (words) => {
-  // Implement this method for wave 4
-};
+    let hand = [];
+    while (hand.length < 10) {
+      hand.push(
+        letterList.pop(Math.floor(Math.random() * (letterList.length - 1)))
+      );
+    }
+
+    return hand;
+  };
+
+  static usesAvailableLetters = (input, lettersInHand) => {
+    // Implement this method for wave 2
+    let hand = [, ...lettersInHand];
+    input = input.toUpperCase();
+    for (const char of input) {
+      if (hand.includes(char) === false) {
```

Comment:
This can be simplified to `if (!hand.includes(char))`. Also, this if check is not necessary since you have the same check on line 60.